File: Part 1 - Data Preprocessing/Data_PreProcessing.R
#installing packages
install.packages("caTools")
library(caTools)

#importing datasets
data=read.csv("Data.csv")

#Missing values
data$Age=ifelse(is.na(data$Age),
                ave(data$Age,FUN = function(x) mean(x,na.rm = TRUE)),
                data$Age)
data$Salary=ifelse(is.na(data$Salary),
                   ave(data$Salary,FUN = function(x) mean(x,na.rm = TRUE)),
                   data$Salary)

#Categorical Data
data$Country=factor(data$Country,
                    levels = c('France','Spain','Germany'),
                    labels = c(1,2,3))

data$Purchased=factor(data$Purchased,
                      levels = c('No','Yes'),
                      labels = c(0,1))

#splitting data in to train and test sets
set.seed(123)
Split=sample.split(data$Purchased,SplitRatio = 0.8)
train_set=subset(data,Split==TRUE)
test_set=subset(data,Split==FALSE)

#feature Scaling
train_set[2:3]=scale(train_set[2:3])
test_set[2:3]=scale(test_set[2:3])

File: Part 1 - Data Preprocessing/Data_PreProcessing.py
#importing lib
import pandas as pd
import numpy as ny
import matplotlib.pyplot as plt

#importing dataset
data=pd.read_csv("Data.csv")
X=data.iloc[:,:-1]
y=data.iloc[:,-1:]

#taking care of missing data
from sklearn.preprocessing import Imputer
imputer=Imputer(missing_values=ny.nan,strategy="mean",axis=0)
imputer=imputer.fit(X.iloc[:,1:3])
X.iloc[:,1:3]=imputer.transform(X.iloc[:,1:3])

#categorical data
from sklearn.preprocessing import LabelEncoder,OneHotEncoder
lab_X=LabelEncoder()
X.iloc[:,0]=lab_X.fit_transform(X.iloc[:,0])
onehot=OneHotEncoder(categorical_features=[0])
X=onehot.fit_transform(X).toarray()

#splitting data into train and test sets
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.25,random_state=0)

#feature scaling
from sklearn.preprocessing import StandardScaler
sc=StandardScaler()
X_train=sc.fit_transform(X_train)
X_test=sc.transform(X_test)